### ISSUE TITLE
HMW4 ready

### ISSUE THREAD

**Issue by juansbr7** [juansbr7/STAT545-hw01-Bonilla-Juan#4]

**Comment by susannaelsie** [comment]
Hi Juan,

I took a look at your hw04! Nice work. Some comments below. 

**Github repository**: I had some trouble finding the correct repository for hw04 (as it is currently housed within your hw01 repo), and then had to navigate through the contents a bit to find your hw04 file (no link in the issue created here and no readme.md file for hw04). This wasn't too hard, but certainly as there are more and more homework files it becomes a bit more difficult to navigate!

**Assignment organization**: It would have been helpful to identify which activity you chose to do right off the bat to assist with my review of the hw so that I could more easily evaluate which operations you performed

**First Activity**
- Super cool first plot! Your use of the facet_grid by year is super helpful (and I have noted it for my own data exploration), and by using the scale_size_continuous by population this figure clearly displays a lot of relevant information. Nice.
- Good job making a tibble with one observation per year and comparing lifeExp in two countries using knitr

**Second Activity**: Looks great! I like your plot that shows patterns of central tendency as well as the range of values for life expectancy by year and by content! Super clear and your approach is simple/elegant and effective!

**Third Activity**:
- Great idea for datasets to merge, providing a clear demonstration of various different join functions and their utility/outcomes
- I'm curious why you created a new object for south america when you should already have it in your environment from the first activity? Not a big thing, I am just trying to pay particular attention to when it is useful to assign names to objects in the code, so I picked up on this.
- in your 'things to notice' section you say that datasets will be joined by either country or official language, but between datasets 2 and 3 I would think you could also use capital city to join them (as another common variable)?

**Process** 
- I like your rationale and approach to exploring data (first looking at the spread of values for gdpPercap and lifeExp in countries across South America). 
- Good reflection at the end about challenges using the gather function. Did you try anything or find any resources to try to figure out how to use gather? 

Overall your assignment was super interesting and I definitely noted a few strategies for plotting data to display multiple variables. Nice job 👍 

Happy Monday,
Susanna

**Comment by derekcho** [comment]
Hi @juansbr7, here are some comments about your hw04:
-	I think that you should bold the Activity (make it the header) and leave the explanation as plain text. This is a small point, but it would make things look nicer!
-	Your first plot with the gdpPercap and lifeExp of South American countries over time is a really cool figure! It makes it easy to compare how this relationship changes over time for each country.
-	Although the scatterplot between Chile and Bolivia is correct, I would not add the geom_smooth(method = “lm”) line since a linear regression line does not have much meaning in this context. It would be more appropriate to add the X=Y line, points above the line indicate that Chile had a higher life expectancy for that year and vice versa. You could also color these by time since it is not immediately obvious which points correspond to which years without consulting the table. 
-	Make sure to include some explanations for each task, not only some of them! Do not just give a bunch of code and plots, we want to see interpretations and explanations as well. What does this particular chunk of code do? What is this figure trying to show? I would rather not have to go through the code line by line to figure out. You plotted a trimmed mean life expectancy, but did not state that it was a trimmed mean rather than a normal mean anywhere and other than in the code! 
-	Your last task was done well! You explained and demonstrated the different types of joins well. However, I think it would have been nice to have shown one or two more types of joins (like outer join).
-	Your coding style is generally quite neat, I can figure out what you are doing when reading the code without too much effort.
-	Your repo could use a quick cleanup! Please put assignments into folders to reduce clutter in your main page.
-	
Your grade will be emailed to you at a later date.

**Comment by ssheikho** [comment]
Hi @juansbr7! I finally got a chance to review your work. I apologize for being so late!

Strength:
+  Your first plot "Life Expectancy in South America" looks great!
+ Good use of  join functions
+ You managed to complete most tasks of this homework

Things to work on:
- I found it hard to navigate through your homework. Maybe add a table of content to your homework, and divide up the homework into (numbered) sections that your toc links to. 
- clean up your repo! it looks really clustered and dense right now. Try organizing each homework into its own folder with a README file explaining the structure of the homework.
- I find it hard to follow code when there is no explanation of what's going on. Try to document your work so others can also understand it!

Good luck :)
